frontend/src/stimulus/controllers/dynamic/admin/statuses.controller.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * -- copyright
+ * OpenProject is an open source project management software.
+ * Copyright (C) the OpenProject GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 3.
+ *
+ * OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+ * Copyright (C) 2006-2013 Jean-Philippe Lang
+ * Copyright (C) 2010-2013 the ChiliProject Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * See COPYRIGHT and LICENSE files for more details.
+ * ++
+ */
+
+import { Controller } from '@hotwired/stimulus';
+
+export default class StatusesController extends Controller {
+  static targets = [
+    'isDefaultCheckbox',
+    'isReadonlyCheckbox',
+  ];
+
+  declare readonly isDefaultCheckboxTarget:HTMLInputElement;
+  declare readonly isReadonlyCheckboxTarget:HTMLInputElement;
+
+  updateReadonlyCheckboxDisabledState() {
+    // do nothing if the readonly checkbox is restricted due to enterprise
+    // edition not being available
+    if (this.isReadonlyCheckboxTarget.dataset.restricted === 'true') {
```

Comment:
💡 It would also be possible to use a [stimulus value](https://stimulus.hotwired.dev/reference/values) to pass in the restricted value explicitly and not via a somewhat nondescript data attribute. I don't know though, if it is possible to have stimulus value attributes defined in places other than where the controller is referenced. Because if it would be necessary to do it directly where the controller is referenced, then it would have to be done in two places (`new/edit.html.erb`). And if it would be necessary to do it in two places, it might not actually be an improvement.

Reply:
That's the thing: the value has to be in the element referencing the controller, and I did not want to duplicate it (as you guessed). That's what I learned when using it in the [progress tracking admin page](https://github.com/opf/openproject/blob/13277cf4bae73205426c94ae658d74838a12779c/app/views/admin/settings/progress_tracking/show.html.erb#L45) where the value of `Setting.work_package_done_ratio` is passed as a value.